### DISCUSSION TITLE
reduce size of prisma docker

### DISCUSSION THREAD

**Discussion by ShadowFox88** [prisma/prisma#25937] [Q&A]
### Question

I'm using the python client for prisma, and whenever i run the command, I end up with around 600MB of just .cache, which is for some reason recreated EVERY time i run the prisma command. How would I go about trying to reduce the image, as just including prisma bumped up that count from ~200MB to ~1.2GB including dependencies. I've tried making builder stages, but it always ends up failing.

### Information about Prisma Schema, Client Queries and Environment (optional)

My Dockerfile:

```docker
FROM python:3.12.8-alpine

ENV POETRY_VIRTUALENVS_CREATE false
ENV PYTHONIOENCODING utf-8

WORKDIR /main

RUN apk add --no-cache gcc python3-dev musl-dev linux-headers openssl git && rm -rf /var/cache/apk/*

ADD poetry.lock .
ADD pyproject.toml .
RUN pip install poetry && poetry install --no-root

ADD ./prisma ./prisma
RUN poetry run prisma generate

ADD . /main

RUN apk del python3-dev musl-dev linux-headers
RUN rm -rf /root/.cache

ENTRYPOINT ["/bin/sh", "-c"]
CMD ["poetry run prisma db push --schema prisma/schema.prisma && python main.py"]
```

**Comment by nurul3101** [comment]
Hello @ShadowFox88 👋 

The Prisma Client Python is community maintained. I would recommend opening this discussion in Prisma Client Python repository [here](https://github.com/RobertCraigie/prisma-client-py) so that the maintainer of Python Client can provide their input.

**Reply by ShadowFox88** [reply]
arguably this is a Prisma question, as I'm installing the Prisma cli that you guys offer
